### PR TITLE
React router project settings

### PR DESCRIPTION
**Pull request by joshsadam**
## Description of changes

Removed `@reach/router` and replace with react router

## Related issue

N/A

## Checklist

Things for the developer to confirm they've done before the PR should be accepted:

*   [ ] ~CHANGELOG.md (and UPGRADING.md if necessary) updated with information for new change.~
*   [ ] ~Tests added (or description of how to test) for any new features.~
*   [ ] ~User documentation updated for UI or technical changes.~

### CODE REVIEW

File: src/main/webapp/resources/js/pages/projects/settings/components/SettingsNav.jsx [hunk 1]
```diff
@@ -1,61 +1,105 @@
...
+                                        /\/projects\/\d+\/settings\/([\w_-]+)/;
+                                      React.useEffect(() => {
+                                        const path = keyRegex.exec(
+                                          location.pathname
+                                        );
+                                        if (path) {
+                                          setKey(path[1]);
+                                        } else {
+                                          throw new Error(
+                                            "Developer: You ed the schema for the settings URL - fix this"
```
Comment (deepsidhu85) [inline src/main/webapp/resources/js/pages/projects/settings/components/SettingsNav.jsx:28]: Might want to change this error message :)

File: src/main/webapp/resources/js/pages/projects/settings/components/metadata/MetadataTemplate.jsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import React from "react";
+import { useParams } from "react-router-dom";
+import { useGetProjectDetailsQuery } from "../../../../../apis/projects/project";
+import MetadataTemplateManager from "./MetadataTemplateManager";
+import MetadataTemplateMember from "./MetadataTemplateMember";
+
```
Comment (deepsidhu85) [inline src/main/webapp/resources/js/pages/projects/settings/components/metadata/MetadataTemplate.jsx:6]: Add file comment

**Review by deepsidhu85 — CHANGES_REQUESTED** [review]
Looking good. Just a few comments below

File: src/main/webapp/resources/js/pages/projects/settings/components/SettingsNav.jsx (hunk 1 above)
Comment (joshsadam) [reply src/main/webapp/resources/js/pages/projects/settings/components/SettingsNav.jsx]: Removed in 123c9e84c9a6ccadd9169144cdc7a975f16b69b6

**Review by deepsidhu85 — APPROVED** [review]

**Pull request merged**